### PR TITLE
Leetcode virtual judge 

### PR DESCRIPTION
**Pull request by shisoft**
This PR is not completed yet. Opened for code review and tracking progress.

### CODE REVIEW

File: src/JoyOI.VirtualJudge.LeetCode/LeetcodePullProblemBodyActor.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+namespace JoyOI.VirtualJudge.LeetCode
+{
+    class LeetcodePullProblemBodyActor
+    {
+        private const string baseUrl = "https://leetcode.com";
+        private const string problemEndpoint = "/problems/{PROBLEM-NAME}/description/";
+        private static List<string> returnFiles = new List<string>(1) { "problem.json" };
+        private static HttpClient client = new HttpClient() { BaseAddress = new Uri(baseUrl) };
+        private static Regex dataRegex = new Regex("var pageData =(?:(?!</script>)[\\s\\S])*");
+        private static Regex bodyRegex = new Regex()
```
Comment (yukozh) [inline src/JoyOI.VirtualJudge.LeetCode/LeetcodePullProblemBodyActor.cs]: I don't think this will pass the compile syntax check

**Review by yukozh — CHANGES_REQUESTED** [review]
Could you fetch the each sub test case details?

**Comment by shisoft** [comment]
@yukozh Leetcode only provide last failed / timeout test case and it have already been attached in `Hint`

File: src/JoyOI.VirtualJudge.LeetCode/LeetcodePullProblemBodyActor.cs (hunk 1 above)
Comment (shisoft) [reply src/JoyOI.VirtualJudge.LeetCode/LeetcodePullProblemBodyActor.cs]: more `\` symbol?

File: src/JoyOI.VirtualJudge.LeetCode/LeetcodePullProblemBodyActor.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+{
+    class LeetcodePullProblemBodyActor
+    {
+        private const string baseUrl = "https://leetcode.com";
+        private const string problemEndpoint = "/problems/{PROBLEM-NAME}/description/";
+        private static List<string> returnFiles = new List<string>(1) { "problem.json" };
+        private static HttpClient client = new HttpClient() { BaseAddress = new Uri(baseUrl) };
+        private static Regex dataRegex = new Regex("var pageData =(?:(?!</script>)[\\s\\S])*");
+        private static Regex bodyRegex = new Regex(@"(?<=<div class=""question-description"">)[\\s\\S]*(?=</div>[\\s\\S]*?<!-- Interview Feedback -->)");
+        private static Regex titleRegex = new Regex(@"(?<=<div class=""question-title clearfix"">[\\s\\S]*?<div class=""row"">[\\s\\S]*?<div class=""col-md-12"">[\\s\\S]*?<h3>[\\s\\S]*?)[\\s\\S]*(?=<\/h3>)");
```
Comment (yukozh) [inline src/JoyOI.VirtualJudge.LeetCode/LeetcodePullProblemBodyActor.cs]: Wrong regex. Should be `(?<=<title>)[a-zA-Z0-9_- ]{1,}(?= - LeetCode</title>)`

**Review by yukozh — CHANGES_REQUESTED** [review]

**Pull request merged**